README.md

```diff
@@ -220,4 +219,29 @@ enum type to int is required. In the factory method, you will receive an integer
 convert it to your enum type and construct the object with it.
 
 Custom type definitions are also validated against a [schema](/schema/custom-codec-schema.json). See the [Schema Validation](#schema-validation) 
-section for details of the validation.
\ No newline at end of file
+section for details of the validation.
+
+### Expanding the Client Protocol
+
+Client protocol can be expanded by adding new
+* services
+* methods
+* parameters to existing method requests, responses or events
+* events to existing methods
+* custom types
+* parameters to existing custom types
+
+While expanding the protocol, one needs to follow these simple guidelines:
+* `since` field of the protocol definitions of the the newly added parameters, methods, events and custom types should 
```

Comment:
Where should we put the current protocol version ? Should it be on README, generator.py or somewhere else ?

Reply:
The latest since tag will define the current version. Maybe a log during code generation will be helpful.

Reply:
I don't think we should hard code else where.